doc/build-osx.md

```diff
@@ -51,6 +51,18 @@ To install, run the following from your terminal:
 brew install automake libtool boost pkg-config libevent
 ```
 
+For macOS 11 (Big Sur) and 12 (Monterey) you need to install a more recent version of llvm.
+
+``` bash
+brew install llvm@14
```

Comment:
> a too recent llvm breaking on the old operating systems

Exactly. Either llvm itself or Homebrew's build scripts might not work for some reason, so this seems safer.

Reply:
Why would LLVM not work? We use the near latest release of LLVM to build the macOS release binaries. There's no point making this LLVM 14 just to "bump it" almost immediately. We should just point to the latest version unless there's an actual reason not to.

Reply:
> Why would LLVM not work?

I have no way to test it. Since Homebrew itself does not support macOS 11 it's reasonable to expect that their newer formulas break, while the old ones might be fine.

This instruction is specifically for old versions of macOS. For newer versions we still have them use the latest thing.

Reply:
I tested this without specifying a llvm version on macos 12.7.4 monterey. It did resolve the compilation errors. I would prefer this without a version, since it is yet another thing to keep track of in our versioning zoo, and I don't think it is preferable for developers to be instructed to use an older compiler. Also not sure if the guide should cover an OS version that is no longer supported by its package manager. If something eventually breaks for a developer, this could be revisited.

Reply:
After a full day my virtual box also managed to complete `brew install llvm` without needing a version :-) Will update the PR tomorrow.